client/app/components/empty-state/EmptyState.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import classNames from 'classnames';
+import { $uibModal } from '@/services/ng';
+import { currentUser } from '@/services/auth';
+import organizationStatus from '@/services/organizationStatus';
+import './empty-state.less';
+
+function createDashboard() {
+  $uibModal.open({
+    component: 'editDashboardDialog',
+    resolve: {
+      dashboard: () => ({ name: null, layout: null }),
+    },
+  });
+}
+
+export function EmptyState({
+  icon,
+  title,
+  description,
+  illustration,
+  helpLink,
+  showAlertStep,
+  showDashboardStep,
+  showInviteStep,
+  onboardingMode,
+}) {
+  const isAdmin = currentUser.isAdmin;
+
+  const dataSourceStepCompleted = organizationStatus.objectCounters.data_sources > 0;
+  const queryStepCompleted = organizationStatus.objectCounters.queries > 0;
+  const dashboardStepCompleted = organizationStatus.objectCounters.dashboards > 0;
+  const alertStepCompleted = organizationStatus.objectCounters.alerts > 0;
+  const inviteStepCompleted = organizationStatus.objectCounters.users > 1;
+
+  const shouldShowOnboarding = !onboardingMode || !(
+    dataSourceStepCompleted &&
+    queryStepCompleted &&
+    dashboardStepCompleted &&
+    inviteStepCompleted
+  );
+
+  if (shouldShowOnboarding) {
+    return (
+      <div className="empty-state bg-white tiled">
+        <div className="empty-state__summary">
+          {title && <h4>{title}</h4>}
+          {icon && <h2><i className={icon} /></h2>}
+          <p>{description}</p>
+
+          <img
+            src={'/static/images/illustrations/' + illustration + '.svg'}
+            alt={illustration + 'Illustration'}
+            width="75%"
+          />
+        </div>
+        <div className="empty-state__steps">
+          <h4>Let&apos;s get started</h4>
+          <ol>
+            <li className={classNames({ done: dataSourceStepCompleted })}>
+              {!isAdmin && <span>Ask an account admin to connect a data source.</span>}
+              {isAdmin && <span><a href="data_sources">Connect</a> a Data Source</span>}
+            </li>
+            <li className={classNames({ done: queryStepCompleted })}>
+              <a href="queries/new">Create</a> your first Query
+            </li>
+            { showAlertStep && (
+              <li className={classNames({ done: alertStepCompleted })}>
+                <a href="alerts/new">Create</a> your first Alert
+              </li>
+            )}
```

Comment:
How about adding an inline component for this:

```jsx
function Step(shouldRender, url, urlText, text, isCompleted) {
  if (!shouldRender) {
    return null;
  }
  return <li className={classNames({ done: isCompleted })}>
    <a href={{url}}>{{urlText}}</a> {{text}}
  </li>;
}
```

And then all these repeated code becomes:

```jsx
<Step shouldRender={showAlertStep} url="alerts/new" urlText="Create" text="your first Alert"/>
```

Also, to avoid repeating so many props, we can define the steps in an array of objects and then just pass `step` to `<Step/>`.